xlsynth-prover/src/enum_in_bound.rs

```diff
@@ -0,0 +1,763 @@
+}
+"#;
+        let mut func = parse_fn(ir_text);
+        let domains = sample_domain();
+        let mut next_id = func.nodes.iter().map(|n| n.text_id).max().unwrap_or(0) + 1;
+        let err = instrument_function_for_enum_bounds(&mut func, &domains, &mut next_id)
+            .expect_err("missing activation should be an error");
+        assert_eq!(
+            err,
+            "Implicit activation parameter not found in IR function '__itok__target'"
+        );
+    }
+
+    const EXPECTED_TARGET_IR: &str = r#"fn __itok__target(__token: token id=1, __activation: bits[1] id=2, x: bits[2] id=3, y: bits[2] id=4) -> (token, bits[2], bits[2]) {
+  sum: bits[2] = add(x, y, id=5)
+  diff: bits[2] = xor(sum, x, id=6)
+  user_fail: bits[1] = literal(value=0, id=7)
+  user_assert: token = assert(__token, user_fail, message="user-level assert", label="user", id=8)
+  enum_in_bound_value____itok__target__x__0: bits[2] = literal(value=0, id=10)
+  enum_in_bound_eq____itok__target__x__0: bits[1] = eq(x, enum_in_bound_value____itok__target__x__0, id=11)
+  enum_in_bound_value____itok__target__x__1: bits[2] = literal(value=1, id=12)
+  enum_in_bound_eq____itok__target__x__1: bits[1] = eq(x, enum_in_bound_value____itok__target__x__1, id=13)
+  enum_in_bound_any____itok__target__x: bits[1] = or(enum_in_bound_eq____itok__target__x__0, enum_in_bound_eq____itok__target__x__1, id=14)
+  enum_in_bound_activation____itok__target__x: bits[1] = and(__activation, enum_in_bound_any____itok__target__x, id=15)
+  enum_in_bound_assert____itok__target__x: token = assert(user_assert, enum_in_bound_activation____itok__target__x, message="Enum argument 'x' out of bounds in function '__itok__target'", label="enum-in-bound::__itok__target::x", id=16)
+  enum_in_bound_value____itok__target__y__0: bits[2] = literal(value=2, id=17)
+  enum_in_bound_eq____itok__target__y__0: bits[1] = eq(y, enum_in_bound_value____itok__target__y__0, id=18)
+  enum_in_bound_value____itok__target__y__1: bits[2] = literal(value=3, id=19)
+  enum_in_bound_eq____itok__target__y__1: bits[1] = eq(y, enum_in_bound_value____itok__target__y__1, id=20)
+  enum_in_bound_any____itok__target__y: bits[1] = or(enum_in_bound_eq____itok__target__y__0, enum_in_bound_eq____itok__target__y__1, id=21)
+  enum_in_bound_activation____itok__target__y: bits[1] = and(__activation, enum_in_bound_any____itok__target__y, id=22)
+  enum_in_bound_assert____itok__target__y: token = assert(enum_in_bound_assert____itok__target__x, enum_in_bound_activation____itok__target__y, message="Enum argument 'y' out of bounds in function '__itok__target'", label="enum-in-bound::__itok__target::y", id=23)
+  ret tuple.9: (token, bits[2], bits[2]) = tuple(enum_in_bound_assert____itok__target__y, diff, sum, id=9)
+}"#;
+
+    const EXPECTED_PROPERTY_IR: &str = r#"fn __enum_in_bound_property____itok__target(__token: token id=24, __activation: bits[1] id=25, x: bits[2] id=26, y: bits[2] id=27) -> bits[1] {
```

Comment:
Expecting exact IR output is likely to be fragile to changes to the way IR is handled. In XLS we'd handle this by putting the expected IR (or verilog) in a separate file which could be programmatically regenerated  if anything broke the test (running with a flag --regenerate_golden_files or something). Fine for now, but if this becomes common we probably want to set up such a system.